src/ethernet_packet.ml

```diff
@@ -25,8 +25,8 @@ module Unmarshal = struct
                                 (get_ethernet_ethertype frame))
       | Some ethertype ->
         let payload = Cstruct.shift frame sizeof_ethernet
-        and source = Macaddr.of_bytes_exn (copy_ethernet_src frame)
-        and destination = Macaddr.of_bytes_exn (copy_ethernet_dst frame)
+        and source = Macaddr.of_octets_exn (copy_ethernet_src frame)
+        and destination = Macaddr.of_octets_exn (copy_ethernet_dst frame)
```

Comment:
is this, even with now ip/macaddr-cstruct the best (i.e. least copying) we can do?

Reply:
it can be improved in the future, but I just want to push the basic changes through that ipaddr-4 needs, not optimise it.  I think we'll need some ppx_cstruct changes and `?off` additions in various *-cstruct libraries to get in-place unmarshalling